src/android/FileTransferBackground.java

```diff
@@ -26,362 +38,379 @@
-    try {
-      upload.put("createdDate", System.currentTimeMillis() / 1000);
-      upload.put("state", UploadState.STARTED);
-      storage.createFile(uploadDirectoryName, fileId + ".json", upload.toString());
-    } catch (Exception e) {
-      e.printStackTrace();
+
+    private ArrayList<String> getOldUploadIds() {
+        Storage storage = SimpleStorage.getInternalStorage(this.cordova.getActivity().getApplicationContext());
+        String uploadDirectoryName = "FileTransferBackground";
+        ArrayList<String> previousUploads = new ArrayList();
+        List<File> files = storage.getFiles(uploadDirectoryName, OrderType.DATE);
+        for (File file : files) {
+            if (file.getName().endsWith(".json")) {
+                String content = storage.readTextFile(uploadDirectoryName, file.getName());
+                if (content != null) {
+                    try {
+                        previousUploads.add(new JSONObject(content).getString("id"));
+                    } catch (JSONException exception) {
+                        logMessage("eventLabel='could not read old uploads' error='" + exception.getMessage() + "'");
+                    }
+                }
+            }
+        }
+        return previousUploads;
     }
-  }
-
-  private void updateStateForUpload(String fileId, String state, String serverResponse) {
-    try {
-      String fileName = fileId + ".json";
-      if (!storage.isFileExist(uploadDirectoryName, fileName)){
-        LogMessage("could not find "+ fileName + " for updating upload info");
-        return;
-      }
-      String content = storage.readTextFile(uploadDirectoryName, fileName);
-      if (content != null) {
-        JSONObject uploadJson = new JSONObject(content);
-        uploadJson.put("state", state);
-        if (state == UploadState.UPLOADED) {
-          uploadJson.put("serverResponse", serverResponse != null ? serverResponse : "");
+
+    private void uploadPendingList() {
+        List<PendingUpload> previousUploads = PendingUpload.all();
+        for (PendingUpload upload : previousUploads) {
+            JSONObject obj = null;
+            try {
+                obj = new JSONObject(upload.data);
+            } catch (JSONException exception) {
+                logMessage("eventLabel='could not parse pending upload' uploadId='" + upload.uploadId + "' error='" + exception.getMessage() + "'");
+            }
+            if (obj != null) {
+                this.upload(obj);
+            }
         }
-        //delete old file
-        removeUploadInfoFile(fileId);
-        //write updated file
-        storage.createFile(uploadDirectoryName, fileName, uploadJson.toString());
-      }
-    } catch (Exception e) {
-      e.printStackTrace();
     }
 
-  }
 
-  private void removeUploadInfoFile(String fileId) {
-    storage.deleteFile(uploadDirectoryName, fileId + ".json");
-  }
+    private void upload(JSONObject jsonPayload) {
```

Comment:
```suggestion
    private void upload(JSONObject jsonPayload) {
      cordova.getThreadPool().execute(new Runnable() {
            public void run() {
```
Cf PR: https://github.com/spoonconsulting/cordova-plugin-background-upload/pull/44